src/core/common/message.cpp

```diff
@@ -143,16 +147,18 @@ void MessagePool::FreeBuffers(Buffer *aBuffer)
     }
 }
 
-otError MessagePool::ReclaimBuffers(int aNumBuffers)
+otError MessagePool::ReclaimBuffers(int aNumBuffers, uint8_t aPriority)
 {
 #if OPENTHREAD_MTD || OPENTHREAD_FTD
     while (aNumBuffers > GetFreeBufferCount())
     {
         MeshForwarder &meshForwarder = GetInstance().GetThreadNetif().GetMeshForwarder();
-        SuccessOrExit(meshForwarder.EvictIndirectMessage());
+        SuccessOrExit(meshForwarder.EvictMessage(aPriority));
```

Comment:
how about passing the gap `(aNumBuffers - GetFreeBufferCount())` into EvictMessage() and ensure evict messages for exact buffers required?

Reply:
I didn't see any benefit to pass the gap (aNumBuffers - GetFreeBufferCount()) into EvictMessage(). Besides of that, it will increase the duplicate code, because both the `mesh_forwarder_ftd.cpp ` and `mesh_forwarder_mtd.cpp`  implement the function `EvictMessage()`.